### DISCUSSION TITLE
[PostgreSQL] Why paramater binding does not work properly with ARRAY?

### DISCUSSION THREAD

**Discussion by VasilyRozhkovVGR** [dbeaver/dbeaver#11664] [Q&A]
#### System information: 
- Win 10 PRO 64 19042.630
- DBeaver 7.2.5.202011152110

#### Connection specification:
- psql (PostgreSQL) 13.0
- PostgreSQL JDBC Driver 

Problems with array parcing. Note: both scripts works well in JetBrain DataGrip
Working code:
![working](https://user-images.githubusercontent.com/74589539/99362836-41e9d580-28b4-11eb-8117-40925e077392.png)
Not working code:
![not_working](https://user-images.githubusercontent.com/74589539/99362846-44e4c600-28b4-11eb-9d3f-5ad11a6bc240.png)

**Comment by uslss** [comment]
Hi @VasilyRozhkovVGR ,
can you provide a script example?

**Comment by VasilyRozhkovVGR** [comment]
-- this code works well (with commented arrays and 'unnest'). i just changed table names.
WITH settings AS (
    	SELECT :start_dt ::timestamp AS start_dt,
           :end_dt ::timestamp   AS end_dt,
           --ARRAY [:site]         AS site,
           :site AS site
),
     vars AS (
         SELECT :app_types as app_types,
                start_dt,
                end_dt
         FROM settings
     ),
     sitegroups AS (
--          SELECT unnest(site) AS site
--          FROM settings
         SELECT site AS site FROM settings
     ),
     apptypes AS (
         SELECT unnest(
                        CASE
                            WHEN app_types = 'mobile' THEN array ['android', 'ios', 'wp']
                            WHEN app_types = 'site'
                                THEN array ['desktop_browser', 'mobile_browser', 'tablet_browser', 'desktop_client']
                            WHEN app_types = 'all'
                                THEN array ['desktop_browser', 'mobile_browser', 'tablet_browser', 'desktop_client',
                                            'android', 'ios', 'wp']
                            END) as app_types
         FROM vars
     ),
     users AS (
         SELECT DISTINCT vurd.uid                                  		uid,
                         vu.sitegroup                                    sitegroup,
                         vu.login                                        login,
                         TO_CHAR(vurd.dt, 'YYYY/MM/DD')                  reg_dt,
                         COALESCE(vud.lastname, '')                      lastname,
                         COALESCE(vud."name", '')                        name_,
                         COALESCE(vud.middlename, '')                    midname
         FROM vurd
                  JOIN vu ON vu.id = vurd.uid 
                  LEFT JOIN vud ON vud.uid = vurd.uid 
         WHERE vurd.app_type IN (SELECT app_types FROM apptypes)
           AND vu.sitegroup IN (SELECT site FROM sitegroups)
         ORDER BY 1
     )
SELECT 
       u.uid,
       u.login,
       u.reg_dt            registration_date,
       u.lastname,
       u.name_,
       u.midname
FROM users u
LIMIT 100;

**Comment by LonwoLonwo** [comment]
Hello @VasilyRozhkovVGR 
I want to ask - if you add a space here in this place (see picture) - then the query works?
![2021-03-09 17_04_39-DBeaver Enterprise 21 0 0 - _none_ Script-9](https://user-images.githubusercontent.com/45152336/110483218-9088dd00-80fa-11eb-8728-6d9c663e23bb.png)

**Comment by LonwoLonwo** [comment]
If you want to use parameters binding inside ARRAY brackets, then you must add a space. Because without a space - this is PostgreSQL [array slicing](https://www.postgresql.org/docs/11/arrays.html#ARRAYS-ACCESSING). We have already fixed it in this ticket - https://github.com/dbeaver/dbeaver/issues/5440

**Comment by VasilyRozhkovVGR** [comment]
> If you want to use parameters binding inside ARRAY brackets, then you must add a space

thnks!!

**Comment by LonwoLonwo** [comment]
Thank you for your update. It looks like the problem was solved. This issue will convert to discussion.
